services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java

```diff
@@ -101,7 +101,8 @@ private static void mergeFiles(String[] truststores, KeyStore truststore, boolea
             if (f.isDirectory()) {
                 mergeFiles(Stream.of(f.listFiles()).map(File::getAbsolutePath).toArray(String[]::new), truststore, false, discoveredFiles);
             } else {
-                if (file.endsWith(".p12") || file.endsWith(".pfx")) {
+                String lowerName = file.toLowerCase();
+                if (KeystoreUtil.KeystoreFormat.PKCS12.getFileExtensions().stream().anyMatch(s -> lowerName.endsWith(s))) {
```

Comment:
It means this condition will be evaluated even in cases like:

- some**p12**
- wrongfile**pfx**
- hello**pkcs12**

Is that correct? As these extensions are set without the dot.

Reply:
Right, this was not intentional with the refactoring. I'll update this to use the new utility method.